tests/Microsoft.Identity.Test.Unit/ManagedIdentityTests/ManagedIdentityTests.cs

```diff
@@ -383,6 +383,51 @@ await mi.AcquireTokenForManagedIdentity(Resource)
             }
         }
 
+        [DataTestMethod]
+        [DataRow(ManagedIdentitySource.AppService, AppServiceEndpoint, HttpStatusCode.NotFound)]
```

Comment:
we should not retry the not found? just a strain on the server. not sure retry will immediately solve the issue, as 404 could be because of various reasons.